### PR TITLE
feat(pkg/osm): Check osm-injector logs for errors

### PR DESCRIPTION
**Pull request by johnsonshi**
feat(pkg/osm): Check osm-injector logs for errors

Checks osm-injector logs for errors and bad logs.
Refactors error detection in osm control plane pod logs.

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

### CODE REVIEW

**Comment by johnsonshi** [comment]
image

File: pkg/osm/pod_logs.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package osm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openservicemesh/osm/pkg/constants"
```
Comment (draychev) [inline pkg/osm/pod_logs.go:7]: This could go to the last section of imports.

**Review by draychev — APPROVED** [review]

File: pkg/osm/pod_logs.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+// Description implements common.Runnable
+func (check NoBadOsmPodLogsCheck) Description() string {
+	return fmt.Sprintf("Checking whether namespace %s has bad (fatal/error/warning/fail) logs in %s pods (container: %s)", check.osmControlPlaneNamespace, check.podName, check.containerName)
+}
+
+// Run implements common.Runnable
+func (check NoBadOsmPodLogsCheck) Run() outcomes.Outcome {
+	//labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"app": constants.OSMControllerName}}
```
Comment (jaellio) [inline pkg/osm/pod_logs.go:70]: Remove comment

File: pkg/utils/http.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package utils
+
+import (
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// IsHTTPResponseCodeEquals checks whether the returned response from the url matches the status code.
+func IsHTTPResponseCodeEquals(url string, statusCode int) error {
```
Comment (jaellio) [inline pkg/utils/http.go:10]: I feel like this method should return a boolean. When "Is" is used in the method name I would expect to get a true or false return value.

**Pull request merged**